crates/accelerate/src/synthesis/permutation/utils.rs

```diff
@@ -144,3 +145,22 @@ pub fn decompose_cycles(cycles: &Vec<Vec<usize>>) -> Vec<(usize, usize)> {
 
     swaps
 }
+
+/// Implements a single swap layer, consisting of conditional swaps between each
+/// neighboring couple. The starting_point is the first qubit to use (either 0 or 1
+///     for even or odd layers respectively). Mutates the permutation pattern ``pattern``.
+pub fn create_swap_layer(
+    pattern: &mut ArrayViewMut1<usize>,
+    starting_point: usize,
+) -> Vec<(usize, usize)> {
+    let num_qubits = pattern.len();
+    let mut gates = Vec::new();
+
+    for j in (starting_point..num_qubits - 1).step_by(2) {
+        if pattern[j] > pattern[j + 1] {
+            gates.push((j, (j + 1)));
+            pattern.swap(j, j + 1);
```

Comment:
it's probably slightly clearer to remove the extra parenthesis around `j+1`, i.e. write `gates.push((j, j+1));`.

Reply:
Without the parenthesis I get an error that it says that 'expected 1 argument and found 2'